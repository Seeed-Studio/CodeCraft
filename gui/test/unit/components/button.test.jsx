import React from 'react';
import {shallow} from 'enzyme';
import ButtonComponent from '../../../src/components/button/button';
import renderer from 'react-test-renderer';

describe('ButtonComponent', () => {
    test('matches snapshot', () => {
        const onClick = jest.fn();
        const component = renderer.create(
            <ButtonComponent onClick={onClick} />
        );
        expect(component.toJSON()).toMatchSnapshot();
    });

    test('triggers callback when clicked', () => {
        const onClick = jest.fn();
        const componentShallowWrapper = shallow(
            <ButtonComponent onClick={onClick} />
        );
        componentShallowWrapper.simulate('click');
        expect(onClick).toHaveBeenCalled();
    });
});
