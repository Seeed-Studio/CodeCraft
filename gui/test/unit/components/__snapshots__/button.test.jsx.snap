// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent matches snapshot 1`] = `
<span
  className=""
  onClick={[Function]}
  role="button"
>
  <div
    className={undefined}
  />
</span>
`;
